Assets/Scripts/NPCs/StateMachine/States/Chase.cs

```diff
@@ -31,21 +38,25 @@ private void Chasing()
 
     private Vector3 ComputeChaseVelocity()
     {
-        Vector3 goalLocation;
-        
+        Vector3 targetLocation;
+
         //The NPC will predict the future position of the player (pursuit behavior)
-        if (_npcManager.GetDistanceWithPlayer() > 1)
+        if (_npcManager.GetDistanceWithPlayer() > NPCsGlobalVariables.ChasePlayerRange) // TODO add raycast here
         {
-            goalLocation = Indestructibles.Player.transform.position + _npcManager.PlayerRigidbody.velocity * NPCsGlobalVariables.ChasePredictionMultiplier;
+            ComputeAStar();
+            targetLocation = _currentTargetLocation;
         }
         //If the NPC is close to the player, just go directly on him
         else
         {
-            goalLocation = Indestructibles.Player.transform.position;
+            _currentTargetLocation = NPCsGlobalVariables.DefaultInitialVector;
```

Comment:
Added this line to reset the A* path when the NPC is chasing the player, is it good for you ?

Reply:
yup